authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -198,34 +194,237 @@ func getActionName(verb string) string {
+
+				| Verb - *        | Resource under specifed apigroups and with all   | Resource under specifed apigroups and with all verbs |
+								  | verbs at cluster scope                           | at ns scope                                          |
+
+				| Group - *       | Resource under all apigroups with specified verb | Resource under all apigroups with specified verb     |
+				|                 | at cluster scope                                 |  at ns scope                                         |
+		*/
+
+		filteredOperations := azureutils.NewOperationsMap()
+		if subRevReq.ResourceAttributes.Group == "*" {
+			// #1 and #3
+			for group, resMap := range operationsMap {
+				if verbMap, found := resMap[subRevReq.ResourceAttributes.Resource]; found {
+					filteredOperations = initializeMapForGroupAndResource(filteredOperations, group, subRevReq.ResourceAttributes.Resource)
+					filteredOperations[group][subRevReq.ResourceAttributes.Resource] = verbMap
+				}
+			}
+		} else { // #2
+			group := "v1" // core api group key
+			if subRevReq.ResourceAttributes.Group != "" {
+				group = subRevReq.ResourceAttributes.Group
+			}
+
+			if resMap, found := operationsMap[group]; found {
+				if verbMap, found := resMap[subRevReq.ResourceAttributes.Resource]; found {
+					filteredOperations = initializeMapForGroupAndResource(filteredOperations, group, subRevReq.ResourceAttributes.Resource)
+					filteredOperations[group][subRevReq.ResourceAttributes.Resource] = verbMap
+				}
+			} else {
+				return nil, errors.Errorf("No resources found for group %s and resource %s", subRevReq.ResourceAttributes.Group, subRevReq.ResourceAttributes.Resource)
+			}
+
+		}
+
+		klog.V(7).Infof("List of filtered operations: %s", filteredOperations)
+
+		// create list of Data Actions
+		authInfoList, err = createAuthorizationActionInfoList(filteredOperations, subRevReq.ResourceAttributes.Verb)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return authInfoList, nil
+}
+
+func initializeMapForGroupAndResource(filteredOperations azureutils.OperationsMap, group string, resourceName string) azureutils.OperationsMap {
+	if _, found := filteredOperations[group]; !found {
+		filteredOperations[group] = azureutils.NewResourceAndVerbMap()
+	}
+
+	if _, found := filteredOperations[group][resourceName]; !found {
+		filteredOperations[group][resourceName] = azureutils.NewVerbAndActionsMap()
+	}
+
+	return filteredOperations
+}
+
+func createAuthorizationActionInfoList(filteredOperations azureutils.OperationsMap, filterVerb string) ([]azureutils.AuthorizationActionInfo, error) {
+	if len(filteredOperations) == 0 {
+		return nil, errors.New("No operations were found for the request.")
+	}
+
+	var authInfos []azureutils.AuthorizationActionInfo
+
+	if filterVerb != "*" {
+		verb := getActionName(filterVerb)
+		for _, resMap := range filteredOperations {
+			for _, resource := range resMap {
```

Comment:
should we rename this to verbsMap for better readability?

Reply:
done